### PR TITLE
Feature new dsl

### PR DESCRIPTION
**Pull request by alemazzo**
- Removed the old DSL
- Introduced the new DSL to create game rules
- Cleanup code & Imports

### CODE REVIEW

**Review by luigi-borriello00 — COMMENTED** [review]
Please add some docs to this magic code :)

File: src/main/scala/scatan/lib/game/dsl/PropertiesDSL.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    def flatMap[R](f: T => Seq[R]): Seq[R] = elem.iterator.flatMap(f).toSeq
+
+  given [P]: Conversion[OptionalProperty[P], MonadProperty[Option[P], P]] with
+    def apply(optionalProperty: OptionalProperty[P]): MonadProperty[Option[P], P] =
+      new MonadProperty(optionalProperty.value)
+
+  given [P]: Conversion[SequenceProperty[P], MonadProperty[Seq[P], P]] with
+    def apply(sequenceProperty: SequenceProperty[P]): MonadProperty[Seq[P], P] =
+      new MonadProperty(sequenceProperty.value)
+
```
Comment (manuandru) [inline src/main/scala/scatan/lib/game/dsl/PropertiesDSL.scala]: Could we use `cats.Monad`?

**Review by manuandru — CHANGES_REQUESTED** [review]

**Review by manuandru — APPROVED** [review]
Next project, we will write a framework for pure functional programming 👽

**Review by luigi-borriello00 — APPROVED** [review]
How many times do you want to change the DSL? YES :)
Btw, great Job!

**Pull request merged**